src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -84,6 +85,13 @@
   private final AzureBlobStorePdao azureBlobStorePdao;
   private final GcsProjectFactory gcsProjectFactory;
 
+  private final Map<UUID, SnapshotProject> snapshotProjects =
```

Comment:
Do we need to set an upper bound?

Reply:
Probably should.

Reply:
I'm not seeing an easy way to set an upper bound on the Map. Did some research, and the `PassiveExpiringMap` is backed by `HashMap`, which can hold ~700 million elements before it blows up. I think we'll need a different caching technology (Redis?) well before we hit that limit, so I'm voting to hold off on setting an upper bound.

Reply:
sold

Reply:
Might be worth checking how big the `HashMap` will get on average so it doesn't lead to memory exhaustion.